### PR TITLE
Ported additions from minkindr that got lost in the package splitting.

### PR DESCRIPTION
**Pull request by gawela**

### CODE REVIEW

**Comment by rdube** [comment]
Tests are running fine on my computer as well. @HannesSommer could that be a Jenkins specific issue? We could not get much more details when looking at Jenkins' console.

**Comment by HannesSommer** [comment]
Sorry, I've overlooked this. Apparently branch fix/gtsam_eigen does not exist any more in eigen_catkin. Jenkins fails on trying to check it out.

**Comment by rdube** [comment]
retest this please

**Comment by rdube** [comment]
@HannesSommer thanks for the tip! Moved the dependency on master for eigen_catkin

**Comment by rdube** [comment]
@HannesSommer would you have a few minutes to review this? It would be nice to pull these features in so that we can stop depending on specific branches! :)

**Comment by rdube** [comment]
or who is supposed to maintain minkindr_gtsam?

File: minkindr_gtsam/CMakeLists.txt [hunk 1]
```diff
@@ -20,7 +20,7 @@ cs_add_library(${PROJECT_NAME}
 
 # add_definitions(-DGTEST_USE_OWN_TR1_TUPLE=0)
 catkin_add_gtest(${PROJECT_NAME}_tests test/test_minkindr-gtsam.cc)
-target_link_libraries(${PROJECT_NAME}_tests ${PROJECT_NAME})
+target_link_libraries(${PROJECT_NAME}_tests ${PROJECT_NAME} pthread)
```
Comment (HannesSommer) [inline minkindr_gtsam/CMakeLists.txt]: This pthread isn't necessary here if gtest is properly installed (~ as source code).

File: minkindr_gtsam/test/test_minkindr-gtsam.cc [hunk 2]
```diff
@@ -28,6 +28,7 @@
 #include <kindr/minimal/cubic-hermite-quaternion-gtsam.h>
 #include <kindr/minimal/common-gtsam.h>
 #include <kindr/minimal/testing-gtsam.h>
+#include "../include/kindr/minimal/cubic-hermite-interpolation-gtsam.h"
 
 #define N_TEST_ITERATIONS 10000
```
Comment (HannesSommer) [inline minkindr_gtsam/test/test_minkindr-gtsam.cc:33]: This looks like a lot. Tests should not take too long. Is it fast enough?

File: minkindr_gtsam/test/test_minkindr-gtsam.cc [hunk 3]
```diff
@@ -683,6 +684,123 @@ TEST(MinkindrGtsamTests, testCubicHermiteQuaternionDerivative) {
...
+  values.insert(3, vaVal);
+  values.insert(4, vbVal);
+
+  EVector3 tA(1), tB(2);
+  EVector3 vA(3), vB(4);
+
+  const double fd_step = 1e-9;
+  const double tolerance = 1e-6;
+
+  {
```
Comment (HannesSommer) [inline minkindr_gtsam/test/test_minkindr-gtsam.cc]: This could benefit from a comment - it isn't that obvious what you are doing here IMO.

**Comment by HannesSommer** [comment]
I'm afraid nobody feels responsible for minkindr_gtsam unless you two do somehow. 

Or is there anybody out there?

Nevertheless I've tried to go over it but with little time and without barking on convention violations.

File: minkindr_gtsam/test/test_minkindr-gtsam.cc (hunk 2 above)
Comment (rdube) [inline minkindr_gtsam/test/test_minkindr-gtsam.cc:33]: Yes the whole testing lasts a couple of seconds.

**Comment by rdube** [comment]
@HannesSommer thanks for the review! I'd suggest to clean this once we further develop / open source this.

**Pull request merged**